File: server/solution/reverse-string/python/SolutionTester.py
import json
import Solution

def stringToIntegerList(input):
    return json.loads(input)

def stringToInt(input):
    return int(input)

def integerListToString(nums, len_of_list=None):
    if not len_of_list:
        len_of_list = len(nums)
    return json.dumps(nums[:len_of_list])

def main():
    with open('testcase.txt', "r") as f:
        lines = f.readlines()
    i = 0
    passall = True
    while i < len(lines) :
        line = lines[i]
        s = line.replace("\n","")
        if (s == "null") :
            s = None
        #print s
        line = lines[i+1]
        #print line
        expected = line.replace("\n","")
        
        ret = Solution.Solution().reverseString(s)

        #print "expected:" + expected
        #print "ret:" + ret
        if (expected == '""' and ret == "") :
            i += 2
            continue

        if (expected != ret) :
            if (s is None) :
                strnums = 'null'
            else:
                strnums = s
            print "[Fail]" + strnums + ";" + ret + ";" + expected
            passall = False
            break

        i = i + 2

    if passall == True :
        print "[Success]Your solution passed all " + str(len(lines)/2) + " test cases!"

if __name__ == '__main__':
    main()

File: server/solution/reverse-string/java/SolutionTester.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collection;

public class SolutionTester {
    private static final String TESTCASE_FILE = "testcase.txt";
    private static final String TESTRESULT_FILE = "testresult.txt";

    public static void main(String[] args) {
        Solution solution = new Solution();
        
        boolean testResult = false;
        try {
            BufferedReader br = new BufferedReader(new FileReader(TESTCASE_FILE));
            try {
                String line;
                int count = 0;
                while ((line = br.readLine()) != null) {
                    // input parameter
                    String s;
                    if (line.equals("null")) {
                        s = null;
                    } else if (line.equals("\"\"")) {
                        s = "";
                    } else {
                        s = line;
                    }
                    // expected result
                    line = br.readLine();
                    String expected;
                    if (line.equals("\"\"")) {
                        expected = "";
                    } else {
                        expected = line;
                    }
                    // create an test execution
                    String ret = solution.reverseString(s);

                    testResult = ret.equals(expected);
                    count++;
                    
                    if (!testResult) {
                        String content = "[Fail]Failed at: Input: " + s + "; Your answer:" + ret + "; Expected answer: " + expected;
                        System.out.println("[Fail]" + s + ";" + ret + ";" + expected);
                        saveTestResult(content);
                        break;
                    }
                }
                if (testResult) {
                    System.out.println("[Success]Your solution passed all " + count +" test cases!");
                    saveTestResult("[Success]Your solution passed all " + count +" test cases!");
                }
            }
            catch (Exception io) {
                //System.out.println(io.getMessage());
                printToFile(io);
            }
            finally {
                br.close();
            }
        }
        catch (IOException ioe) {
            printToFile(ioe);
        } finally {
        }
    }

    public static void saveTestResult(String content) {
        BufferedWriter bw = null;
		try {
		    bw = new BufferedWriter(new FileWriter(TESTRESULT_FILE));
			bw.write(content);
		} catch (IOException ioe) {
			ioe.printStackTrace();
		} finally {
			try {
				if (bw != null) {
					bw.close();
                }
			} catch (IOException ex) {
				ex.printStackTrace();
			}
		}
    }

    public static void printToFile(Exception ex) {
        PrintWriter pw = null;
        try {
		    pw = new PrintWriter(TESTRESULT_FILE);
            ex.printStackTrace(pw);
		} catch (IOException ioe) {
			ioe.printStackTrace();
		} finally {
			try {
				if (pw != null) {
					pw.close();
                }
			} catch (Exception ex2) {
				ex2.printStackTrace();
			}
		}
    }
}
